client/session.go

```diff
@@ -2213,14 +2245,35 @@ func (s *session) streamBlocksMetadataFromPeerV2(
 
 			clonedID := idPool.BinaryID(data)
 
+			var (
+				tags         ident.Tags
+				tagDecodeErr error
+			)
+			if encodedTags := elem.EncodedTags; len(encodedTags) != 0 {
+				tagDecoderCheckedBytes.Reset(encodedTags)
+				tagDecoder.Reset(tagDecoderCheckedBytes)
+				tags = make(ident.Tags, 0, tagDecoder.Remaining())
+				for tagDecoder.Next() {
+					curr := tagDecoder.Current()
+					tags = append(tags, idPool.CloneTag(curr))
+				}
+				tagDecodeErr = tagDecoder.Err()
+			}
+
 			// Error occurred retrieving block metadata, use default values
+			var err error
 			if elem.Err != nil {
```

Comment:
maybe a comment that element err takes precedence over tag decoding error

Reply:
Sure thing.